modules/slugs/includes/class-qtranslate-slug.php

```diff
@@ -108,17 +108,15 @@ public function get_meta_key( $force_lang = false ) {
     }
 
     /**
-     * Check if exists qtranslate and do the installation, support multisite.
+     * Do the installation, support multisite.
      */
     public function install() {
-        global $wpdb;
-
-        if ( function_exists( 'is_multisite' ) && is_multisite() ) {
-            if ( isset( $_GET['networkwide'] ) && ( $_GET['networkwide'] == 1 ) ) {
-                $old_blog = $wpdb->blogid;
-                $blogids  = $wpdb->get_col( $wpdb->prepare( "SELECT blog_id FROM $wpdb->blogs" ) );
-                foreach ( $blogids as $blog_id ) {
-                    switch_to_blog( $blog_id );
+        if ( is_multisite() ) {
+            if ( is_plugin_active_for_network(plugin_basename(QTRANSLATE_FILE)) ) {
+                $old_blog = get_current_blog_id();
+                $blogs  = wp_get_sites();
+                foreach ( $blogs as $blog ) {
+                    switch_to_blog( $blog[blog_id]);
```

Comment:
`blog_id` is not defined. I suppose you meant `$blog->id`. Not tested apparently.

Reply:
my bad, could not test multisite with my current test environment... this just looked like straight forward.
`wp_get_sites()` should return arrays with key `blog_id`, while `get_sites()` is returning `WP_Site` arrays with property `blog_id`. Agree and changed to the latter